server/signalhttp/instrumentation.go

```diff
@@ -0,0 +1,82 @@
+package signalhttp
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type instrumentationMiddleware struct {
+	requestCounter  *prometheus.CounterVec
+	requestSize     *prometheus.SummaryVec
+	requestDuration *prometheus.HistogramVec
+	responseSize    *prometheus.HistogramVec
+}
+
+// HandlerInstrumenter can instrument handlers.
+type HandlerInstrumenter interface {
+	NewHandler(labels prometheus.Labels, handler http.Handler) http.HandlerFunc
```

Comment:
Naming: maybe we can call this func `Instrument` since that is the action it is doing, and it matches the interface convention as well